prow/config/tide_test.go

```diff
@@ -33,33 +34,83 @@ import (
 var testQuery = TideQuery{
 	Orgs:                   []string{"org"},
 	Repos:                  []string{"k/k", "k/t-i"},
+	ExcludedRepos:          []string{"org/repo"},
 	Labels:                 []string{labels.LGTM, labels.Approved},
 	MissingLabels:          []string{"foo"},
 	Author:                 "batman",
 	Milestone:              "milestone",
 	ReviewApprovedRequired: true,
 }
 
+var expectedQueryComponents = []string{
+	"is:pr",
+	"state:open",
+	"archived:false",
+	"label:\"lgtm\"",
+	"label:\"approved\"",
+	"-label:\"foo\"",
+	"author:\"batman\"",
+	"milestone:\"milestone\"",
+	"review:approved",
+}
+
 func TestTideQuery(t *testing.T) {
 	q := " " + testQuery.Query() + " "
-	checkTok := func(tok string) {
-		if !strings.Contains(q, " "+tok+" ") {
-			t.Errorf("Expected query to contain \"%s\", got \"%s\"", tok, q)
-		}
-	}
+	checkTok := checkTok(t, q)
```

Comment:
I am btw inclined to change the tests to simply test for a literal search string. It has the drawback that ordering will matter so if someone changes the code in a way that the query is the same but with a different order, those tests will fail incorrectly but:
* It makes the tests much easier to understand (Expected query to be X, was Y)
* It also verifies that there are no additional incorrect parts in the search string, this may introduce bugs that will go unnoticed with the current tests (I had a bug around that initially)

Reply:
Does anyone have objections to that?

Reply:
okay the static string doesn't work because the result string depends on a map order and that is not deterministic. I have added a check for duplicates though